### PR TITLE
Switch OTLP exporter to internal data format

### PR DESCRIPTION
**Pull request by dmitryax**
This commit switches OTLP exporter to use internal data format instead of OC which makes it possible to use internal data for the whole OTLP based pipeline.

### CODE REVIEW

File: service/builder/pipelines_builder.go [hunk 1]
```diff
@@ -259,19 +259,22 @@ func createTraceProcessor(
...
-		return factoryBase.(component.ProcessorFactoryOld).CreateTraceProcessor(logger, nextConsumerOld, cfg)
+		return factoryOld.CreateTraceProcessor(logger, nextConsumerOld, cfg)
 	}
 
-	// Old type processor and a new type consumer usecase is not supported.
-	// TODO: This case can be supported since we have OC->internal traces translation function
-	return nil, errors.New("OC Traces -> internal data format translation is not supported")
+	// If processor is of the old type, but downstream consumer is of the new type,
+	// use NewInternalToOCTraceConverter compatibility shim to convert traces from internal format to OC.
+	traceConverter := converter.NewOCToInternalTraceConverter(nextConsumer.(consumer.TraceConsumer))
```
Comment (tigrannajaryan) [inline service/builder/pipelines_builder.go:276]: 👍

File: testbed/testbed/load_generator.go [hunk 2]
```diff
@@ -220,6 +227,55 @@ func (lg *LoadGenerator) generateTrace() {
...
 
+func (lg *LoadGenerator) generateTrace() {
+	traceSender := lg.sender.(TraceDataSender)
+
+	itemsPerBatch := int(lg.options.ItemsPerBatch)
+
+	traceData := data.NewTraceData()
+	traceData.ResourceSpans().Resize(1)
+	traceData.ResourceSpans().At(0).InstrumentationLibrarySpans().Resize(1)
+	spans := traceData.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans()
```
Comment (tigrannajaryan) [inline testbed/testbed/load_generator.go:238]: Can we use temp vars to avoid repetition and make this more readable?

Comment (dmitryax) [reply testbed/testbed/load_generator.go]: updated

**Comment by dmitryax** [comment]
@tigrannajaryan PTAL

**Comment by dmitryax** [comment]
@tigrannajaryan @bogdandrutu @pjanotti could you take a look and merge if looks good

**Pull request merged**